### ISSUE TITLE
nvcc_wrapper doesn't work with tooling in development environments like CLion

### ISSUE THREAD

**Issue by dhollman** [kokkos/kokkos#3296]
Here's an example of the kind of error messages I get:

```
Cannot get compiler information:
	Compiler exited with error code 1: /ascldap/users/dshollm/Projects/kokkos/source/develop/bin/nvcc_wrapper -xc++ -DGTEST_HAS_PTHREAD=0 -DGTEST_HAS_TR1_TUPLE=0 -I/ascldap/users/dshollm/Projects/kokkos/build/devel/clion_kokkos-dev-2-nvcc-10.1_debug/algorithms/unit_tests -I/tmp/tmp.e88RDNaqJ4/algorithms/unit_tests -I/tmp/tmp.e88RDNaqJ4/algorithms/unit_tests/../src -I/tmp/tmp.e88RDNaqJ4/tpls/gtest -Wno-unused-command-line-argument -Wall -Wno-shadow -pedantic -Wsign-compare -Wtype-limits -Wuninitialized -Wno-format -g -std=c++14 -fpch-preprocess -v -dD -E
	nvcc fatal   : No input files specified; use option --help for more information
```

**Comment by DavidPoliakoff** [comment]
Yeah, I've seen exactly this error. I wound up having to work around nvcc_wrapper in my attempts (just calling NVCC directly). This would be a really neat feature if we could make it work

**Comment by calewis** [comment]
@jjwilke has some experience in compiler wrappers.

**Comment by jjwilke** [comment]
Where is that command coming from?  I don't see how calling `nvcc` directly would get rid of the error.

**Comment by crtrott** [comment]
yeah I don't see how that would work with nvcc other than Clion having hardcoded executable detection and it will only work with compilers it explicitly knows about.

**Comment by DavidPoliakoff** [comment]
It works with NVCC, I've made that work with Kokkos

**Comment by DavidPoliakoff** [comment]
CLion invokes the compiler to get some info out of it, apparently that invocation trips us up

**Comment by jjwilke** [comment]
How does CLion handle compiler launchers in CMake?

**Comment by jjwilke** [comment]
This problem might go away with the new PR

**Comment by DavidPoliakoff** [comment]
Very well might, @dhollman have you tried? I can if not

**Comment by jjwilke** [comment]
@DavidPoliakoff I'm also curious about nvcc. The command above just appears to doing a basic query of the C++ compiler, but nvcc chokes on the flags above.  What exactly are you doing when you say "it works with NVCC"?

**Comment by DavidPoliakoff** [comment]
My situation might be a bit different than Daisy's. I've given up on CLion to develop Kokkos, so I install Kokkos normally. But then I have an app. I have two paths

1. nvcc_wrapper
2. CUDA as a language, use NVCC

I can't make nvcc_wrapper and CLion see eye to eye, I get errors like what Daisy mentions. When I just compile with CUDA as a lanaguage and use NVCC as the compiler, everything with CLion is happy and works (I get autocomplete, errors, all that good stuff).

**Comment by crtrott** [comment]
Shouldn't CLion have something which honors the cmake config of the project? I mean in stuff like VS and Eclipse I can tell it all the configure stuff from cmake and it will do the right thing.

**Comment by DavidPoliakoff** [comment]
@crtrott, CLion does an invocation of the compiler outside of CMake to determine something about it (I forget details, it has been a long time since I've done this, now I just don't use nvcc_wrapper when I'm using CLion). Something in that invocation isn't honored by nvcc_wrapper, I believe.

**Comment by crtrott** [comment]
Just make nvcc_wrapper dump all the arguments to it, and then you should know what it is and can fix nvcc_wrapper.

**Comment by dhollman** [comment]
Yeah, I've done this before. The issue is basically that we're not forwarding some flags through the `nvcc_wrapper` correctly, especially in the case where there's no source file. It's not hard to fix, just messy.

**Comment by ajpowelsnl** [comment]
@crtrott, @dalg24 -- can we close this issue?

**Comment by dalg24** [comment]
The issue is still there with Kokkos 3.7(ish) and CLion 2022.1

Note that this is a failure to retrieve information about the compiler, not a failure to configure.  This means you can still build and run executables.
I agree that it is rather irritating but I don't know that I care to fix.

Note that with a recent NVC++ (no `nvcc_wrapper` involved) I get
```
Cannot get compiler information:
	Compiler exited with error code 1: /opt/nvidia/hpc_sdk/Linux_x86_64/22.9/compilers/bin/nvc++ -xc++ -DKOKKOS_DEPENDENCE -isystem/opt/nvidia/hpc_sdk/Linux_x86_64/22.9/cuda/11.7/include -g -O0 -gpu=nordc -cuda -gpu=cc80 -mp --c++17 --gnu_extensions -fpch-preprocess -v -dD -E
	nvc++-Error-Unknown switch: -xc++
	nvc++-Error-Unknown switch: -fpch-preprocess
